app/src/main/java/de/tum/in/tumcampusapp/activities/StartupActivity.java

```diff
@@ -147,6 +148,34 @@ protected void onCreate(Bundle savedInstanceState) {
         //Show a loading screen during boot
         setContentView(R.layout.activity_startup);
 
+        // init easter egg (logo)
+        ImageView tumLogo = findViewById(R.id.startup_tum_logo);
+        if (Utils.getSettingBool(this, Const.RAINBOW_MODE, false)) {
+            tumLogo.setImageResource(R.drawable.tum_logo_rainbow);
+        } else {
+            tumLogo.setImageResource(R.drawable.tum_logo);
+        }
+
+        tapCounter = 0;
+        View background = findViewById(R.id.startup_background);
+        background.setOnClickListener(view -> {
+            tapCounter++;
+            if(tapCounter == 3){
```

Comment:
This OnClockListener in Startup- / MainActivity is duplicated code. Please extract the common logic